corehq/apps/data_dictionary/templates/data_dictionary/base.html

```diff
@@ -141,7 +141,7 @@ <h4 class="modal-title">{% trans "Create a new Case Type" %}</h4>
     </p>
   </div>
   {% if not request.is_view_only %}
-    <div data-bind="saveButton: saveButton, visible: $root.activeCaseType()"></div>
+    <div id="save-button" data-bind="saveButton: saveButton, visible: $root.activeCaseType()"></div>
```

Comment:
do we want to try to make this unique across all of HQ now or just this page itself?
So, should this be `save-data-dictionary-button`? The way these events are tracked, we are 100% sure it would for the relevant page only?

Reply:
We are using a combination of page path along with the element id so that the identification is for the relevant page only.
However I agree the suggested name is much better to avoid chance of false positives from inherited htmls.
Addressed in 7b667e84048c4ec59ad46acae1ad0b943285a1e0

Reply:
Cool.

Does the same apply for other IDs like `upload-dict`? I think we could also use an abbreviation `dd` and not write `data-dictionary` everywhere.

Reply:
> Does the same apply for other IDs like upload-dict? 
Yes thats correct. Combining page path check along with id makes chances of false positives close to 0. 

> I think we could also use an abbreviation dd and not write data-dictionary everywhere.
That is a good idea. Feel like we should use this as prefix for all ids to guarantee uniqueness. Sounds a good idea ?

Reply:
:+1:

Reply:
Addressed ee38a1db913b625260fc1e8f5bf3cb5c114a9bf7

Reply:
These don't need to be namespaced, I'd go with the original name for the save of brevity.

Once you add events, it's important that their properties, label, etc. (the parameters passed to tracking functions like [this](https://github.com/dimagi/commcare-hq/blob/e90635f650487e1ea3f6fcbc67dae9064e519945/corehq/apps/analytics/static/analytix/js/google.js#L118)) be unique across HQ. But this element id only need to be unique on the page (not because of analytics, all HTML id should be unique within their page).

> We are using a combination of page path along with the element id so that the identification is for the relevant page only.

What does this mean?

Reply:
> But this element id only need to be unique on the page (not because of analytics, all HTML id should be unique within their page)

Hey @orangejenny 
Just adding some context here & @ajeety4 can correct me if I get this wrong since he is much more involved in integration with google analytics.
The way we track events, we mark it for a certain URL along with the ID of element.
If the URL tracking is faulty and matches irrelevant pages unknowingly, we would get false events for the ID if it exists in two places that matched the URL. A rare case I agree. Hence, the extra safety to consider keeping ID as unique as possible across HQ. Like 'save-btn' would be present on many pages.

Reply:
Hey @mkangia what you said is correct.

Hey @orangejenny , you are right about tracking through console.
To provide more context, for this ticket we are using both Google Tag Manager and Kissmetrics events console to track events directly without making code changes. We use `id` to identify the element being clicked, so if the same id is present on a different page it will create false positive.However using page path also as an additional property to identify the element prevents this.
An example below from Kissmetrics:
![image](https://github.com/dimagi/commcare-hq/assets/125016806/7e4d2727-4082-4455-86d4-a3b7204457bb)


**Naming:**
Apart from safety gurard that Manish mentioned, figured having the name of feature (data-dict) could avoid accidental (rare) repeat with names from the inherited templates. (especially for generic names -  `save-btn`).
Let me know if that makes sense.

> https://github.com/dimagi/commcare-hq/pull/34554 has a sendEvent function

Good query here. This sendEvent function is currently being used only sending user properties to GTM and could potentially be used to track something that is not possible from GTM console.

> It'd also be helpful in understanding when we should be using GTM versus kiss.

GTM is only a tag manager and hence can be used to insert any tracking code (gtag installation or even kiss installation script) dynamically. For analytics, it must be used in tandem with a tool like Google Analytics where events would be sent out.

In terms of dynamic tracking through console (without code), GTM and [kiss](https://support.kissmetrics.io/docs/tracking-rules-event-manager) both support this however GTM has far more advanced capabilities in terms of what can be tracked and identification of the elements. One example would be `css-selector` support in GTM.
Kissmetrics requires unique css class (single supported) or id to identify the element. GTM does have a steep learning curve compared to kiss.

Reply:
@ajeety4 That all makes sense, thank you for the thorough explanation. I get why you're namespacing the ids now, and why we're adding this functionality in the first place. Two things I'm a little worried about:
* Making id names unique across HQ sounds onerous.
* Having dependencies outside of the code base on the code is a little weird. At some point someone is going to delete an id after grepping for it and finding nothing.

These are both pretty minor, but one idea to mitigate them is to have a namespace for these kinds of ids. So instead of `dd-create-case-type-btn`, something like `gtm-create-case-type-btn`. It'd be short, it'd be a known convention (eventually) and it'd be pretty easy to grep for duplicates.

Reply:
@orangejenny  Agreed on the points mentioned. Having the flexibility of tracking events through console does comes with some drawbacks.

Thank you for the idea - that sounds good to me. Having such namespaces `gtm-` or `kiss-` would be a good indicator of  these elements being used externally. I will also update the Readme in that regards.

@zandre-eng pinging you here since we have similar work going on.

Reply:
Addressed in 349b11bd412a391d198ad8fbf43aff8cf9c9c270